docs/source/howto/data.rst

```diff
@@ -276,7 +276,52 @@ Sharing data
 Deleting data
 =============
 
-`#3999`_
+By default, every time you run or submit a new calculation or workflow, AiiDA will create for you new nodes in the database, and will never replace or delete data.
+There are cases, however, when it might be useful to delete nodes that are not useful anymore, for instance test runs or incorrect/wrong data and calculations.
+In this case, AiiDA provides a ``verdi node delete`` command to remove the nodes from the provenance graph.
+
+.. note::
+   The ``verdi node delete`` command should be used with great care.
+
+   First, once the data is deleted, there is no way to recover it (unless you made a backup).
+
+   Even more critically, even if you ask to delete only one node, ``verdi node delete`` will typically delete a number of additional linked nodes, in order to preserve a consistent state of the provenance graph.
+   For instance, if you delete an input of a calculation, AiiDA will delete also the calculation itself (as otherwise you would be effectively changing the inputs to that calculation in the provenance graph).
+   The full set of consistency rules are explained in detail :ref:`here <topics:provenance:consistency>`.
+
+   Therefore: always check the output of ``verdi node delete`` to make sure that it is not deleting more than you expect.
+   You can also use the ``--dry-run`` flag of ``verdi node delete`` to see what the command would do without performing any actual operation.
+
+In addition, there are a number of additional rules that are not mandatory to ensure consistency, but can be toggled by the user.
```

Comment:
Is this referring to the text in the "note" block? Usually one should be able to assume that the flow of the text is not interrupted if one omits reading the extra blocks.

Reply:
Indeed - I also had the same doubt. I now left only a minimal note/caution block, and moved the content outside. The other option was to move everything in the block but it seemed less elegant.